src/core/thread/mle_types.hpp

```diff
@@ -567,6 +568,61 @@ class LeaderData : public otLeaderData
     void SetLeaderRouterId(uint8_t aRouterId) { mLeaderRouterId = aRouterId; }
 };
 
+OT_TOOL_PACKED_BEGIN
+class RouterIdSet
```

Comment:
Would a more natural location for this be in `router_table.hpp`?

Reply:
I also prefer the definition in `router_table.hpp`, but I think it may cause include dependency loops. If not easy to move it, I'd say ok to keep it in `mle_type.hpp`.

Reply:
yes, like @abtink mentioned, putting it here was to avoid include dependency loops.